pkgs/by-name/hy/hyprlock/package.nix

```diff
@@ -0,0 +1,80 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libGL
+, libxkbcommon
+, hyprlang
+, pam
+, wayland
+, wayland-protocols
+, cairo
+, pango
+, libdrm
+, mesa
+, nix-update-script
+, expat
+, libXdmcp
+, pcre2
+, util-linux
+, libselinux
+, libsepol
+, pcre
+, fribidi
+, libthai
+, libdatrie
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hyprlock";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = "hyprlock";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-SX3VRcewkqeAIY6ptgfk9+C6KB9aCEUOacb2pKl3kO0=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    cairo
+    expat
+    fribidi
```

Comment:
Do we really need newly added dependencies?

Reply:
I went through and added them as the build errored or threw warnings. AFAICT they are all required